enhancements/support-for-realtime-kernel.md

```diff
@@ -0,0 +1,293 @@
+---
+title: support-for-real-time-kernels
+authors:
+  - "@miabbott"
+reviewers:
+  - "@ashcrow"
+  - "@cgwalters"
+  - "@crawford"
+  - "@darkmuggle"
+  - "@davidvossel"
+  - "@hardys"
+  - "@imcleod"
+  - "@jlebon"
+  - "@MarSik"
+  - "@mrguitar"
+  - "@runcom"
+  - "@simon3z"
+  - "@sinnykumari"
+  - "@stbenjam"
+approvers:
+  - "@ashcrow"
+  - "@cgwalters"
+  - "@crawford"
+  - "@imcleod"
+  - "@runcom"
+creation-date: 2019-12-19
+last-updated: 2019-12-19
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Support For Real-time Kernels
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Many workloads associated with telcos and other verticals like the financial
+service industry (FSI) require a high degree of determinism.  While Linux is
+not a real-time operating system, the real-time kernel provides a preemptive
+scheduler providing the OS with real-time characteristics. We want to provide
+customers the ability to select the real-time kernel for Red Hat Enterprise
+Linux CoreOS (RHCOS) nodes in the cluster.
+
+## Motivation
+
+The real-time kernel, along with realtime-friendly hardware and BIOS settings,
+is a requirement to meet the low latency needs that Verizon, Ericsson, Nokia,
+along with other network equipment providers (NEP), are looking to deploy as
+part of the 5G roll-out.
+
+Unfortunately, the additional overhead from the scheduler makes it a poor fit
+for the majority of “general purpose” workloads, therefore, real-time is not
+suited to be the default kernel in RHCOS. OpenShift Container Platform (OCP)
+clusters and MachineConfigPools will need to be configurable to run either kernel.
+
+### Goals
+
+- Provide the `kernel-rt` packages in the `machine-os-content` image
+- Provide the ability to select the real-time kernel for RHCOS nodes in the cluster
+- Provide initial tuning of the RHCOS nodes for real-time workloads
+
+### Non-Goals
+
+- Defaulting to the real-time kernel or offering multiple boot images
+- Additional tuning of the RHCOS nodes after the real-time kernel is selected (handled by [Cluster Node Tuning Operator](https://github.com/openshift/cluster-node-tuning-operator))
+- Providing real-time kernel support to older versions (pre-4.4) of OCP
+- Supporting real-time kernel selection on non-RHCOS nodes
+
+## Proposal
+
+1.  Include the `kernel-rt` packages in the `machine-os-content` image
+2.  Provide tunable in MachineConfig that selects the type of kernel to use
+3.  Do initial tuning of RHCOS node after real-time kernel is applied
```

Comment:
I think we need to clarify what "tuning" is affected. For example, FIPS and encryption?

Reply:
I believe this falls into the bucket of `tuned` profiles.  @cgwalters can you provide some details?